doc/nrf/libraries/modem/location.rst

```diff
@@ -0,0 +1,198 @@
+
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_NRF_CLOUD`
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_HERE` and :kconfig:`CONFIG_MULTICELL_LOCATION_HERE_API_KEY` (see below other authentication options)
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_SKYHOOK` and :kconfig:`CONFIG_MULTICELL_LOCATION_SKYHOOK_API_KEY`
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_POLTE` and :kconfig:`CONFIG_MULTICELL_LOCATION_POLTE_CUSTOMER_ID` and :kconfig:`CONFIG_MULTICELL_LOCATION_POLTE_API_TOKEN`
+
+For WiFi location services, use at least one of the following sets of options and configure corresponding authentication parameters:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_NRF_CLOUD`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE` and :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_API_KEY`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK` and :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_API_KEY`
+
+Following WiFi service related options can usually have default values:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_HOSTNAME`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_TLS_SEC_TAG`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_HOSTNAME`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_TLS_SEC_TAG`
+
+Usage
+*****
+
+Using Location library is rather easy. First you need to initialize the library with :c:func:`location_init`.
+
+Secondly, you need to set configuration (:c:struct:`loc_config`) including location method configurations (:c:struct:`loc_method_config`).
+This is achieved easily by setting first default values by using :c:func:`loc_config_defaults_set` and c:func:`loc_config_method_defaults_set`,
+and then setting any required non-default values to the structures.
+
+Once the configuration is set up properly, you just need to call c:func:`location_request` with the configuration.
+
+Samples using the library
+*************************
+
+The following |NCS| samples use this library:
+
+* :ref:`location_sample`
+* :ref:`modem_shell_application`
+
+Limitations
+***********
+
+* Location library can only have one client registered at a time. If there is already a client handler registered, another initialization will override the existing handler.
+* Cellular neighbor information used for cellular positioning is more accurate on Modem FW (MFW) 1.3.0 compared to earlier MFW releases which do not have API for scanning the neighboring cells.
+  For older than MFW 1.3.0 releases, only serving cell information is provided and that information may be from hours or days ago, or even older, depending on modem sleep states.
+
+Dependencies
+************
+
+This library uses the following |NCS| libraries:
+
+* :ref:`nrf_modem_lib_readme`
+* :ref:`lte_lc_readme`
+* :ref:`lib_multicell_location`
+* :ref:`lib_rest_client`
+* :ref:`lib_nrf_cloud`
+* :ref:`lib_nrf_cloud_agps`
+* :ref:`lib_nrf_cloud_pgps`
+* :ref:`lib_nrf_cloud_rest`
+
```

Comment:
modem_info and date_time libraries missing

Reply:
Added all these.

Reply:
Actually cannot find proper target to link for the cjson_lib because it's in modules and not NRF or Zephyr.